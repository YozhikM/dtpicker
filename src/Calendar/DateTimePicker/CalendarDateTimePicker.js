/* @flow */

import React from 'react';
import SvgIcon from '../../SvgIcon/SvgIcon';
import CalendarDate from '../DateTimePicker/CalendarDate';
import DateMaskInput from '../DateMaskInput';
import s from './Calendar.scss';

type Props = {
  value: Date,
  visibleDate?: Date,
  highlight: Array<Date> | Date,
  onChange?: Date => void,
  onSetDate?: Date => void,
  onSetDateByClick?: Date => void,
  time?: boolean,
  icon?: boolean,
  leftArrow?: boolean,
  rightArrow?: boolean,
  isCalendarShown?: boolean,
};

type State = {
  highlight?: Array<Date> | Date,
  value: Date,
  visibleDate: Date,
  isCalendarShown: boolean,
};

class CalendarDateTimePicker extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      visibleDate: this.props.visibleDate || new Date(),
      value: this.props.value,
      isCalendarShown: !!this.props.isCalendarShown,
    };
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.visibleDate !== this.state.visibleDate)
      this.setState({ visibleDate: nextProps.visibleDate });

    if (nextProps.value !== this.state.value) {
      this.setState({ value: nextProps.value });
    }

    if (nextProps.isCalendarShown !== this.state.isCalendarShown) {
      this.setState({ isCalendarShown: nextProps.isCalendarShown });
    }
  }

  onChangeInputValue = (date: Date) => {
    const { onChange, onSetDate } = this.props;

    if (date && onSetDate) {
      onSetDate(date);
    }
    if (date && onChange) {
      onChange(date);
    }
  };

  onChange = (value: Date) => {
    const { onChange } = this.props;

    if (onChange) {
      onChange(value);
    }
  };

  onSetDate = (date: Date) => {
    const { onSetDateByClick } = this.props;

    if (onSetDateByClick) onSetDateByClick(date);
  };

  onSetTime = (time: Date) => {
    this.setState({
      visibleDate: time,
    });
  };

  render() {
    const {
      highlight,
      icon,
      leftArrow,
      rightArrow,
      time,
      isCalendarShown,
      visibleDate,
    } = this.props;
    const { value } = this.state;
    const style = {
      width: '80%',
    };

    return (
      <div className={s.container}>
        <DateMaskInput style={style} visibleDate={visibleDate} onChange={this.onChangeInputValue} />
        {icon && (
          <div className={s.icon}>
            <SvgIcon file="calendar" />
          </div>
        )}
        {isCalendarShown && (
          <CalendarDate
            highlight={highlight}
            value={value}
            visibleTime={visibleDate}
            onSetDate={this.onSetDate}
            onSetTime={this.onSetTime}
            onChange={this.onChange}
            leftArrow={leftArrow}
            rightArrow={rightArrow}
            time={time}
          />
        )}
      </div>
    );
  }
}

export default CalendarDateTimePicker;
