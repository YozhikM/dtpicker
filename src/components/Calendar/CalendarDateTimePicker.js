/* @flow */

import React from 'react';
import CalendarDate from '../Calendar/CalendarDate';
import MaskedInput from 'react-text-mask';
import SvgIcon from '../SvgIcon';
import s from './Calendar.scss';
import { format, isValid, isSameDay } from 'date-fns';

type Props = {|
  highlight: Array<Date> | Date,
  value: Date,
  visibleDate?: Date,
  time?: boolean,
  onChangeCalendarVisibility?: boolean => void,
  isCalendarShown?: boolean,
  onChange?: (value: Date) => void,
  borderLeft?: boolean,
  borderRight?: boolean,
  icon?: boolean,
  onSetDate: (date: Date) => void,
  leftArrow?: boolean,
  rightArrow?: boolean,
|};
type State = {|
  highlight: Array<Date> | Date,
  value: Date,
  visibleDate: Date,
  isCalendarShown: boolean,
  inputValue: string,
|};

class CalendarDateTimePicker extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      highlight: this.props.highlight,
      visibleDate: this.props.visibleDate,
      value: this.props.value,
      isCalendarShown: !!this.props.isCalendarShown,
      inputValue: '',
    };
  }

  formatWithTime = (date: Date) => {
    return format(date, 'DD/MM/YYYY HH:mm');
  };
  formatWithoutTime = (date: Date) => {
    return format(date, 'DD/MM/YYYY');
  };

  componentWillMount() {
    const { visibleDate, time } = this.props;
    if (time) {
      this.setState({ inputValue: this.formatWithTime(visibleDate) });
    } else {
      this.setState({
        inputValue: this.formatWithoutTime(visibleDate),
      });
    }
  }

  componentWillReceiveProps(nextProps: Props) {
    const { time } = this.props;
    if (nextProps.time !== time) {
      if (time) {
        this.setState({
          inputValue: this.formatWithTime(nextProps.visibleDate),
        });
      } else {
        this.setState({
          inputValue: this.formatWithoutTime(nextProps.visibleDate),
        });
      }
    }
    if (
      nextProps.visibleDate !== this.state.visibleDate &&
      !isSameDay(nextProps.visibleDate, this.props.value)
    ) 
		this.setState({ visibleDate: nextProps.visibleDate });
	{
      if (time) {
        this.setState({
          inputValue: this.formatWithTime(nextProps.visibleDate),
        });
      } else {
        this.setState({
          inputValue: this.formatWithoutTime(nextProps.visibleDate),
        });
      }
    }

    if (nextProps.value !== this.state.value) {
      this.setState({ value: nextProps.value });
    }

    if (nextProps.isCalendarShown !== this.state.isCalendarShown) {
      this.setState({ isCalendarShown: nextProps.isCalendarShown });
    }
  }

  onChangeInputValue = (e: any) => {
    const { onChange, onSetDate, time } = this.props;
    const value = e.target.value;
    this.setState({ inputValue: value });
    let userDate;
    const YY = Number(value.substr(6, 4));
    const MM = Number(value.substr(3, 2)) - 1;
    const DD = Number(value.substr(0, 2));
    const HH = Number(value.substr(11, 2));
    const mm = Number(value.substr(14, 2));
    if (time) {
      userDate = new Date(YY, MM, DD, HH, mm);
    } else {
      userDate = new Date(YY, MM, DD);
    }
    if (value.length > 0 && onSetDate && isValid(userDate)) {
      onSetDate(userDate);
      onChange(userDate);
    }
  };

  onFocusInput = () => {
    const { isCalendarShown } = this.state;
    if (!isCalendarShown) {
      this.onChangeVisibility();
    }
  };

  onPressEnter = (e: Event) => {
    if (e.key === 'Enter') {
      this.onChangeVisibility();
    }
  };

  onChangeVisibility = () => {
    const { isCalendarShown } = this.state;
    const { onChangeCalendarVisibility } = this.props;
    if (onChangeCalendarVisibility) {
      onChangeCalendarVisibility(!isCalendarShown);
    }
  };

  onChange = (value: Date) => {
    const { onChange } = this.props;
    if (onChange) {
      onChange(value);
    }
  };

  onSetDate = (date: Date) => {
    const { time } = this.props;
    if (time) {
      this.setState({
        inputValue: this.formatWithTime(date),
      });
    } else {
      this.setState({
        inputValue: this.formatWithoutTime(date),
      });
    }
    const { onSetDate } = this.props;
    if (onSetDate) onSetDate(date);
  };

  onSetTime = (time: Date) => {
    this.setState({
      visibleDate: time,
      inputValue: this.formatWithTime(time),
    });
  };

  render() {
    const {
      highlight,
      borderLeft,
      borderRight,
      icon,
      leftArrow,
      rightArrow,
      time,
      isCalendarShown,
      visibleDate,
	  isSingleCalendar
    } = this.props;
    const { inputValue, value } = this.state;
    const iconStyle = {
      display: icon ? 'flex' : 'none',
    };
    const borderStyle = {
      borderLeft: borderLeft ? '3px solid #34495e' : 0,
      borderRight: borderRight ? '3px solid #34495e' : 0,
    };
    let mask;
    if (time) {
      mask = [
        /[0-9]/,
        /\d/,
        '/',
        /\d/,
        /\d/,
        '/',
        /\d/,
        /\d/,
        /\d/,
        /\d/,
        ' ',
        /\d/,
        /\d/,
        ':',
        /\d/,
        /\d/,
      ];
    } else {
      mask = [
        /[0-9]/,
        /\d/,
        '/',
        /\d/,
        /\d/,
        '/',
        /\d/,
        /\d/,
        /\d/,
        /\d/,
      ];
    }

    return (
      <div className={s.container}>
        <MaskedInput
          mask={mask}
          className={s.input}
          style={borderStyle}
          type="text"
          onChange={this.onChangeInputValue}
          onFocus={this.onFocusInput}
          onKeyPress={this.onPressEnter}
          value={inputValue}
          showMask
        />
        <div
          onClick={this.onChangeVisibility}
          className={s.icon}
          style={iconStyle}
        >
          <SvgIcon file="calendar" />
        </div>
        {isCalendarShown && (
          <CalendarDate
            highlight={highlight}
            value={value}
            visibleTime={visibleDate}
            onSetDate={this.onSetDate}
            onSetTime={this.onSetTime}
            onChange={this.onChange}
            leftArrow={leftArrow}
            rightArrow={rightArrow}
            time={time}
			isSingleCalendar={isSingleCalendar}
          />
        )}
      </div>
    );
  }
}

export default CalendarDateTimePicker;
