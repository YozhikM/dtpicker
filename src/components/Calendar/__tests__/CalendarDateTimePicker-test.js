/* @flow */

import React from 'react';
import CalendarDateTimePicker from '../CalendarDateTimePicker';
import MaskedInput from 'react-text-mask';

describe('CalendarDateTimePicker', () => {
  describe('render()', () => {
    it('basic', () => {
      const date = new Date(2017, 9, 8);
      expect(
        shallow(
          <CalendarDateTimePicker
            highlight={date}
            visibleDate={date}
            value={new Date(2001, 3)}
            onSetDate={jest.fn()}
          />
        )
      ).toMatchSnapshot();
    });

    it('if !time', () => {
      const date = new Date(2017, 9, 8);
      expect(
        shallow(
          <CalendarDateTimePicker
            highlight={date}
            visibleDate={date}
            value={new Date(2001, 3)}
            onSetDate={jest.fn()}
          />
        )
      ).toMatchSnapshot();
    });

    it('if isCalendarShown', () => {
      expect(
        shallow(
          <CalendarDateTimePicker
            highlight={new Date(2017, 9, 8)}
            visibleDate={new Date(2017, 9, 8)}
            value={new Date(2001, 3)}
            onSetDate={jest.fn()}
            isCalendarShown
          />
        )
      ).toMatchSnapshot();
    });
  });

  describe('componentWillMount()', () => {
    it('if !time', () => {
      const highlight = new Date(2009, 2, 4);
      const wrapper = shallow(
        <CalendarDateTimePicker
          highlight={highlight}
          visibleDate={highlight}
          value={new Date()}
          onSetDate={jest.fn()}
        />
      );
      const format = wrapper.instance().formatWithoutTime(highlight);
      expect(wrapper.state('inputValue')).toBe(format);
    });

    it('if time', () => {
      const highlight = new Date(2010, 2, 4);
      const wrapper = shallow(
        <CalendarDateTimePicker
          highlight={highlight}
          visibleDate={highlight}
          value={new Date()}
          onSetDate={jest.fn()}
          time
        />
      );
      const format = wrapper.instance().formatWithTime(highlight);
      expect(wrapper.state('inputValue')).toBe(format);
    });
  });

  it('formatWithTime()', () => {
    const wrapper = shallow(
      <CalendarDateTimePicker
        highlight={new Date(2017, 9, 8)}
        visibleDate={new Date()}
        value={new Date()}
        onSetDate={jest.fn()}
      />
    );
    const newDate = new Date(2017, 10, 10);
    const format = wrapper.instance().formatWithTime(newDate);
    expect(format).toBe('10/11/2017 00:00');
  });

  it('formatWithoutTime()', () => {
    const wrapper = shallow(
      <CalendarDateTimePicker
        highlight={new Date(2017, 9, 8)}
        visibleDate={new Date()}
        value={new Date()}
        onSetDate={jest.fn()}
      />
    );
    const newDate = new Date(2009, 9, 1);
    const format = wrapper.instance().formatWithoutTime(newDate);
    expect(format).toBe('01/10/2009');
  });

  describe('componentWillReceiveProps()', () => {
    describe('time', () => {
      it('if time', () => {
        const highlight = new Date(2099, 1, 0);
        const wrapper = shallow(
          <CalendarDateTimePicker
            highlight={highlight}
            visibleDate={highlight}
            value={new Date(2009, 3, 1)}
            onSetDate={jest.fn()}
            time={false}
          />
        );
        wrapper.setProps({ time: true });
        const format = wrapper.instance().formatWithTime(highlight);
        expect(wrapper).toMatchSnapshot();
        expect(format).toBe('31/01/2099 00:00');
      });

      it('if !time', () => {
        const highlight = new Date(2099, 1, 0);
        const wrapper = shallow(
          <CalendarDateTimePicker
            highlight={highlight}
            visibleDate={highlight}
            value={new Date(2009, 3, 1)}
            onSetDate={jest.fn()}
            time
          />
        );
        wrapper.setProps({ time: false });
        const format = wrapper.instance().formatWithoutTime(highlight);
        expect(wrapper).toMatchSnapshot();
        expect(format).toBe('31/01/2099');
      });
    });

    describe('visibleDate', () => {
      it('if !time', () => {
        const wrapper = shallow(
          <CalendarDateTimePicker
            highlight={new Date(2009, 2, 4)}
            visibleDate={new Date(2009, 2, 4)}
            value={new Date(2004, 5, 7)}
            onSetDate={jest.fn()}
          />
        );
        const newDate = new Date(2000, 10, 11);
        wrapper.setProps({ visibleDate: newDate });
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.state('inputValue')).toBe('11/11/2000');
      });

      it('if time', () => {
        const wrapper = shallow(
          <CalendarDateTimePicker
            highlight={new Date(2010, 1, 11)}
            visibleDate={new Date(2010, 1, 11)}
            value={new Date(2011, 9, 7)}
            onSetDate={jest.fn()}
            time
          />
        );
        const newDate = new Date(2010, 0, 10);
        wrapper.setProps({ visibleDate: newDate });
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.state('inputValue')).toBe('10/01/2010 00:00');
      });
    });

    it('value', () => {
      const wrapper = shallow(
        <CalendarDateTimePicker
          highlight={new Date(2060, 5, 11)}
          visibleDate={new Date(2060, 5, 11)}
          value={new Date(2051, 3, 1)}
          onSetDate={jest.fn()}
        />
      );
      const newDate = new Date(2010, 0, 10);
      wrapper.setProps({ value: newDate });
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.state('value')).toBe(newDate);
    });

    it('isCalendarShown', () => {
      const wrapper = shallow(
        <CalendarDateTimePicker
          highlight={new Date(2060, 5, 11)}
          visibleDate={new Date(2060, 5, 11)}
          value={new Date(2051, 3, 1)}
          onSetDate={jest.fn()}
          isCalendarShown={false}
        />
      );
      wrapper.setProps({ isCalendarShown: true });
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.state('isCalendarShown')).toBe(true);
    });
  });

  it('onChangeVisibility()', () => {
    const spy = jest.fn();
    const wrapper = shallow(
      <CalendarDateTimePicker
        highlight={new Date()}
        visibleDate={new Date()}
        value={new Date()}
        onSetDate={jest.fn()}
        onChangeCalendarVisibility={spy}
      />
    );
    wrapper.instance().onChangeVisibility();
    expect(spy).toBeCalledWith(true);
  });

  it('onChange()', () => {
    const spy = jest.fn();
    const wrapper = shallow(
      <CalendarDateTimePicker
        highlight={new Date()}
        visibleDate={new Date()}
        value={new Date()}
        onSetDate={jest.fn()}
        onChange={spy}
        isCalendarShown
      />
    );
    const newDate = new Date(2019, 8, 3);
    wrapper
      .find('CalendarDate')
      .props()
      .onChange(newDate);
    expect(spy).toBeCalledWith(newDate);
  });

  it('onSetDate()', () => {
    const spy = jest.fn();
    const wrapper = shallow(
      <CalendarDateTimePicker
        highlight={new Date()}
        visibleDate={new Date()}
        value={new Date()}
        onSetDate={spy}
        isCalendarShown
      />
    );
    const newDate = new Date(2009, 2, 3);
    wrapper
      .find('CalendarDate')
      .props()
      .onSetDate(newDate);
    expect(spy).toBeCalledWith(newDate);
    expect(wrapper.state('inputValue')).toBe('03/03/2009');
  });

  it('onSetTime()', () => {
    const wrapper = shallow(
      <CalendarDateTimePicker
        highlight={new Date()}
        visibleDate={new Date()}
        value={new Date()}
        onSetDate={jest.fn()}
        isCalendarShown
      />
    );
    const newDate = new Date('2019-12-12T18:00:00.000Z');
    wrapper
      .find('CalendarDate')
      .props()
      .onSetTime(newDate);
    expect(wrapper.state('inputValue')).toBe('13/12/2019 00:00');
    expect(wrapper.state('visibleDate')).toEqual(newDate);
  });

  describe('user interactions', () => {
    it('onPressEnter()', () => {
      const spy = jest.fn();
      const wrapper = shallow(
        <CalendarDateTimePicker
          highlight={new Date(2009, 1, 10)}
          visibleDate={new Date(2009, 1, 10)}
          value={new Date(1994, 4, 1)}
          onSetDate={jest.fn()}
          onChangeCalendarVisibility={spy}
        />
      );
      const input = wrapper.find(MaskedInput);
      input.simulate('keyPress', { keyCode: 13 });
      expect(wrapper.state('isCalendarShown')).toBe(false);
    });

    it('onFocusInput()', () => {
      const spy = jest.fn();
      const wrapper = shallow(
        <CalendarDateTimePicker
          highlight={new Date(2009, 1, 10)}
          visibleDate={new Date(2009, 1, 10)}
          value={new Date(1994, 4, 1)}
          onSetDate={jest.fn()}
          onChangeCalendarVisibility={spy}
        />
      );
      const input = wrapper.find(MaskedInput);
      input.simulate('focus');
      expect(wrapper).toMatchSnapshot();
      expect(spy).toBeCalledWith(true);
    });

    it('onChangeInputValue()', () => {
      const spy = jest.fn();
      const spy2 = jest.fn();
      const wrapper = shallow(
        <CalendarDateTimePicker
          highlight={new Date(2009, 1, 10)}
          visibleDate={new Date(2009, 1, 10)}
          value={new Date(2111, 5, 3)}
          onSetDate={spy}
          onChange={spy2}
        />
      );
      const input = wrapper.find(MaskedInput);
      input.simulate('change', { target: { value: '10/12/2019 12:00' } });
      expect(wrapper).toMatchSnapshot();
      const newDate = new Date('2019-12-09T18:00:00.000Z');
      expect(spy).toBeCalledWith(newDate);
      expect(spy2).toBeCalledWith(newDate);
    });
  });
});
