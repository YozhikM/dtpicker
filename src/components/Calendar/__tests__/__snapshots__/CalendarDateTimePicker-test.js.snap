// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDateTimePicker componentWillReceiveProps() isCalendarShown 1`] = `
<div>
  <t
    mask={
      Array [
        /\\[0-9\\]/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    showMask={true}
    style={
      Object {
        "borderLeft": 0,
        "borderRight": 0,
      }
    }
    type="text"
    value="11/06/2060"
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <SvgIcon
      file="calendar"
    />
  </div>
  <CalendarDate
    highlight={2060-06-10T18:00:00.000Z}
    onChange={[Function]}
    onSetDate={[Function]}
    onSetTime={[Function]}
    value={2051-03-31T18:00:00.000Z}
    visibleTime={2060-06-10T18:00:00.000Z}
  />
</div>
`;

exports[`CalendarDateTimePicker componentWillReceiveProps() time if !time 1`] = `
<div>
  <t
    mask={
      Array [
        /\\[0-9\\]/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    showMask={true}
    style={
      Object {
        "borderLeft": 0,
        "borderRight": 0,
      }
    }
    type="text"
    value="31/01/2099 00:00"
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <SvgIcon
      file="calendar"
    />
  </div>
</div>
`;

exports[`CalendarDateTimePicker componentWillReceiveProps() time if time 1`] = `
<div>
  <t
    mask={
      Array [
        /\\[0-9\\]/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        " ",
        /\\\\d/,
        /\\\\d/,
        ":",
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    showMask={true}
    style={
      Object {
        "borderLeft": 0,
        "borderRight": 0,
      }
    }
    type="text"
    value="31/01/2099"
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <SvgIcon
      file="calendar"
    />
  </div>
</div>
`;

exports[`CalendarDateTimePicker componentWillReceiveProps() value 1`] = `
<div>
  <t
    mask={
      Array [
        /\\[0-9\\]/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    showMask={true}
    style={
      Object {
        "borderLeft": 0,
        "borderRight": 0,
      }
    }
    type="text"
    value="11/06/2060"
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <SvgIcon
      file="calendar"
    />
  </div>
</div>
`;

exports[`CalendarDateTimePicker componentWillReceiveProps() visibleDate if !time 1`] = `
<div>
  <t
    mask={
      Array [
        /\\[0-9\\]/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    showMask={true}
    style={
      Object {
        "borderLeft": 0,
        "borderRight": 0,
      }
    }
    type="text"
    value="11/11/2000"
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <SvgIcon
      file="calendar"
    />
  </div>
</div>
`;

exports[`CalendarDateTimePicker componentWillReceiveProps() visibleDate if time 1`] = `
<div>
  <t
    mask={
      Array [
        /\\[0-9\\]/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        " ",
        /\\\\d/,
        /\\\\d/,
        ":",
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    showMask={true}
    style={
      Object {
        "borderLeft": 0,
        "borderRight": 0,
      }
    }
    type="text"
    value="10/01/2010 00:00"
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <SvgIcon
      file="calendar"
    />
  </div>
</div>
`;

exports[`CalendarDateTimePicker render() basic 1`] = `
<div>
  <t
    mask={
      Array [
        /\\[0-9\\]/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    showMask={true}
    style={
      Object {
        "borderLeft": 0,
        "borderRight": 0,
      }
    }
    type="text"
    value="08/10/2017"
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <SvgIcon
      file="calendar"
    />
  </div>
</div>
`;

exports[`CalendarDateTimePicker render() if !time 1`] = `
<div>
  <t
    mask={
      Array [
        /\\[0-9\\]/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    showMask={true}
    style={
      Object {
        "borderLeft": 0,
        "borderRight": 0,
      }
    }
    type="text"
    value="08/10/2017"
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <SvgIcon
      file="calendar"
    />
  </div>
</div>
`;

exports[`CalendarDateTimePicker render() if isCalendarShown 1`] = `
<div>
  <t
    mask={
      Array [
        /\\[0-9\\]/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    showMask={true}
    style={
      Object {
        "borderLeft": 0,
        "borderRight": 0,
      }
    }
    type="text"
    value="08/10/2017"
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <SvgIcon
      file="calendar"
    />
  </div>
  <CalendarDate
    highlight={2017-10-07T18:00:00.000Z}
    onChange={[Function]}
    onSetDate={[Function]}
    onSetTime={[Function]}
    value={2001-03-31T17:00:00.000Z}
    visibleTime={2017-10-07T18:00:00.000Z}
  />
</div>
`;

exports[`CalendarDateTimePicker user interactions onChangeInputValue() 1`] = `
<div>
  <t
    mask={
      Array [
        /\\[0-9\\]/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    showMask={true}
    style={
      Object {
        "borderLeft": 0,
        "borderRight": 0,
      }
    }
    type="text"
    value="10/12/2019 12:00"
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <SvgIcon
      file="calendar"
    />
  </div>
</div>
`;

exports[`CalendarDateTimePicker user interactions onFocusInput() 1`] = `
<div>
  <t
    mask={
      Array [
        /\\[0-9\\]/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        "/",
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
        /\\\\d/,
      ]
    }
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    showMask={true}
    style={
      Object {
        "borderLeft": 0,
        "borderRight": 0,
      }
    }
    type="text"
    value="10/02/2009"
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <SvgIcon
      file="calendar"
    />
  </div>
</div>
`;
